pxtblocks/code-validation/validateCommentsExist.ts

```diff
@@ -0,0 +1,20 @@
+
+namespace pxt.blocks {
+    // validates that one or more blocks comments are in the project
+    // returns the blocks that have comments for teacher tool scenario
+    export function validateBlockCommentsExist({ usedBlocks }: { usedBlocks: Blockly.Block[] }): Blockly.Block[] {
+        const commentBlocks = usedBlocks.filter((block) => !!block.getCommentText());
+        return commentBlocks;
```

Comment:
For similar reasons that we accept an object as the parameter, I think it'd make sense to return an object as well, so the returned information has a built-in label. And I also think having a "passed: boolean" field on that object would be very helpful. I wish we'd added one to the blocksExistValidator too, but that's not something to worry about in this change.

Ditto on the function below.